src/urllib3/connection.py

```diff
@@ -229,6 +229,11 @@ def putheader(self, header, *values):
             )
 
     def request(self, method, url, body=None, headers=None):
+        # Update the inner socket's timeout value to send the request.
+        # This only triggers if the connection is re-used.
+        if self.sock is not None:
```

Comment:
To support AppEngine and other HTTPConnection implementations we can add `if getattr(self, "sock", None) is not None:`

Reply:
Done